modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/DeleteStoreQueryRuntime.java

```diff
@@ -0,0 +1,124 @@
+
+    private String queryName;
+    private MetaStreamEvent.EventType eventType;
+    private QuerySelector selector;
+    private StateEventPool stateEventPool;
+    private MetaStreamEvent metaStreamEvent;
+    private Attribute[] outputAttributes;
+
+    public DeleteStoreQueryRuntime(String queryName, MetaStreamEvent metaStreamEvent) {
+        this.queryName = queryName;
+        this.eventType = metaStreamEvent.getEventType();
+        this.metaStreamEvent = metaStreamEvent;
+        this.setOutputAttributes(metaStreamEvent.getLastInputDefinition().getAttributeList());
+    }
+
+    @Override
+    public Event[] execute() {
+        try {
+            StateEvent stateEvent = new StateEvent(1, outputAttributes.length);
+            StreamEvent streamEvent = new StreamEvent(metaStreamEvent.getBeforeWindowData().size(),
+                    metaStreamEvent.getOnAfterWindowData().size(),
+                    metaStreamEvent.getOutputData().size());
+            stateEvent.addEvent(0, streamEvent);
+
+            ComplexEventChunk complexEventChunk = new ComplexEventChunk(stateEvent, stateEvent, true);
+
+            if (eventType == MetaStreamEvent.EventType.TABLE) {
+                selector.process(complexEventChunk);
+            }
+            return new Event[]{};
+        } catch (Throwable t) {
+            throw new StoreQueryRuntimeException("Error executing '" + queryName + "', " + t.getMessage(), t);
+        }
+    }
+
+    @Override
+    public void reset() {
+        if (selector != null) {
+            selector.process(generateResetComplexEventChunk(metaStreamEvent));
+        }
+    }
+
+    public void setStateEventPool(StateEventPool stateEventPool) {
+        this.stateEventPool = stateEventPool;
+    }
+
+    /**
+     * This method sets selector for the delete store query runtime.
+     *
+     * @param selector for the store query
+     */
+    public void setSelector(QuerySelector selector) {
+        this.selector = selector;
+    }
+
+    /**
+     * This method sets the output attribute list of the given store query.
+     *
+     * @param outputAttributeList
+     */
+    public void setOutputAttributes(List<Attribute> outputAttributeList) {
+        this.outputAttributes = outputAttributeList.toArray(new Attribute[outputAttributeList.size()]);
+    }
+
+    @Override
+    public Attribute[] getStoreQueryOutputAttributes() {
+        return Arrays.copyOf(outputAttributes, outputAttributes.length);
+    }
+
+    private ComplexEventChunk<ComplexEvent> generateResetComplexEventChunk(MetaStreamEvent metaStreamEvent) {
+        StreamEvent streamEvent = new StreamEvent(metaStreamEvent.getBeforeWindowData().size(),
+                metaStreamEvent.getOnAfterWindowData().size(), metaStreamEvent.getOutputData().size());
+        streamEvent.setType(ComplexEvent.Type.RESET);
+
+        StateEvent stateEvent = stateEventPool.borrowEvent();
+        if (eventType == MetaStreamEvent.EventType.AGGREGATE) {
```

Comment:
Will this happen

Reply:
In this case, it won't. 
Fixed with commit a9c25b9